From f64aede7de3f2a8fdaf0db144ce0eb14fbc1a68c Mon Sep 17 00:00:00 2001
From: Federico Beffa <address@hidden>
Date: Tue, 20 Jan 2015 18:43:55 +0100
Subject: [PATCH] gnu: libpeas: Update to 1.12.1, add arguments and disable
 tests.

* gnu/packages/gnome.scm (libpeas): Update to version 1.12.1, add 'pre-build
  and 'check-after-install phases and delete 'check phase.  Given that
  'check-after-install only solves part of the problem, disable the tests.
---
 gnu/packages/gnome.scm | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/gnu/packages/gnome.scm b/gnu/packages/gnome.scm
index acc964e..7622e0a 100644
--- a/gnu/packages/gnome.scm
+++ b/gnu/packages/gnome.scm
@@ -3,6 +3,7 @@
 ;;; Copyright © 2014 Ludovic Courtès <address@hidden>
 ;;; Copyright © 2014 Ian Denhardt <address@hidden>
 ;;; Copyright © 2014 Eric Bavier <address@hidden>
+;;; Copyright © 2014, 2015 Federico Beffa <address@hidden>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -427,7 +428,7 @@ some form of information without getting in the user's way.")
 (define-public libpeas
   (package
     (name "libpeas")
-    (version "1.9.0")
+    (version "1.12.1")
     (source
      (origin
       (method url-fetch)
@@ -436,7 +437,7 @@ some form of information without getting in the user's way.")
                           name "-" version ".tar.xz"))
       (sha256
        (base32
-        "13fzyzv6c0cfdj83z1s16lv8k997wpnzyzr0wfwcfkcmvz64g1q0"))))
+        "1mjjjjwphc83bjznmbsm7x0jg7ql261nys6qnl7mi0nkr4qvw476"))))
     (build-system gnu-build-system)
     (inputs
      `(("atk" ,atk)
@@ -449,6 +450,21 @@ some form of information without getting in the user's way.")
        ("glib:bin" ,glib "bin")
        ("gobject-introspection" ,gobject-introspection)
        ("intltool" ,intltool)))
+    (arguments
+     ;; Some tests make use of libtool convenience libraries.  This, together
+     ;; with our patched 'gobject-introspection', results in temporary
+     ;; .gir/.typelib files referring to non-existent directories.  The
+     ;; installed files do not include those entries.
+     `(#:tests? #f
+       #:phases 
+       (alist-cons-before
+        'build 'pre-build
+        (lambda* _
+          (setenv "CC" "gcc"))
+        (alist-cons-after
+         'install 'check-after-install
+         (assq-ref %standard-phases 'check)
+         (alist-delete 'check %standard-phases)))))
     (home-page "https://wiki.gnome.org/Libpeas")
     (synopsis "GObject plugin system")
     (description
@@ -457,7 +473,6 @@ every application the chance to assume its own extensibility.  It also has a
 set of features including, but not limited to: multiple extension points; on
 demand (lazy) programming language support for C, Python and JS; simplicity of
 the API")
-
     (license license:lgpl2.0+)))
 
 (define-public gtkglext
-- 
1.8.4

